probe-rs-t2rust/src/lib.rs

```diff
@@ -0,0 +1,310 @@
+use std::fs::{read_dir, read_to_string, File};
+use std::io::{self, Write};
+use std::path::{Path, PathBuf};
+
+pub fn run(input_dir: impl AsRef<Path>, output_file: impl AsRef<Path>) {
+    let mut f = File::create(output_file.as_ref()).unwrap();
+
+    // Determine all config files to parse.
+    let mut files = vec![];
+    visit_dirs(input_dir.as_ref(), &mut files).unwrap();
+
+    let mut configs: Vec<proc_macro2::TokenStream> = vec![];
+    for file in files {
+        let string = read_to_string(&file).expect(
+            "Algorithm definition file could not be read. This is a bug. Please report it.",
+        );
+
+        let yaml: Result<serde_yaml::Value, _> = serde_yaml::from_str(&string);
+
+        match yaml {
+            Ok(chip) => {
+                let chip = extract_chip_family(&chip);
+                configs.push(chip);
+            }
+            Err(e) => {
+                panic!("Failed to parse target file: {:?} because:\n{}", file, e);
+            }
+        }
+    }
+
+    let use_statements = r#"
```

Comment:
I think this all would be nicer if it was in the quote! macro instead of a string.